conn.go

```diff
@@ -172,7 +172,7 @@ func Connect(addr string, cfg *ConnConfig, errorHandler ConnErrorHandler, sessio
 		r:            bufio.NewReader(conn),
 		cfg:          cfg,
 		uniq:         make(chan int, streams),
-		calls:        make([]callReq, streams),
+		calls:        make([]*callReq, streams), // TODO(zariel): use a sparse map instead?
```

Comment:
We would have to protect the map with a mutex though...  so we can do it like this if we don't want to have to lock.